update vendor for new test changes and vendoring (#1834)
openshift/api: README.md

			mustRegister()
```

### defining tests
Tests are logically associated with FeatureGates.
When adding any FeatureGated functionality a new test file is required.

> The test files are located in `<group>/<version>/tests/<crd-name>/FeatureGate.yaml`
> ```
> route
>   v1
>     tests
>       routes.route.openshift.io

Reviewer: Punctuation can make it a little more obvious that the ```` ``` ````-block represents a filesystem hierarchy:
````suggestion
The test files are located in `<group>/<version>/tests/<crd-name>/FeatureGate.yaml`:
```
route/
  v1/
    tests/
      routes.route.openshift.io/
````

        AAA_ungated.yaml
        ExternalRouteCertificate.yaml
```

> 
> Here's an `AAA_ungated.yaml` example
> ```yaml
> apiVersion: apiextensions.k8s.io/v1 # Hack because controller-gen complains if we don't have this
> name: "Route"

Reviewer: I'm a fan of proper punctuation for English prose.  However, the double-quotes around "Route" are conspicuous, especially as none of the other values are quoted in the yaml—are the double-quotes actually needed?
````suggestion
Here's an `AAA_ungated.yaml` example:
```yaml
apiVersion: apiextensions.k8s.io/v1 # Hack because controller-gen complains if we don't have this.
name: Route
````

crdName: routes.route.openshift.io
tests:
```

> Here's an `ExternalRouteCertificate.yaml` example
> ```yaml
> apiVersion: apiextensions.k8s.io/v1 # Hack because controller-gen complains if we don't have this
> name: "Route"

Reviewer: ````suggestion
Here's an `ExternalRouteCertificate.yaml` example:
```yaml
apiVersion: apiextensions.k8s.io/v1 # Hack because controller-gen complains if we don't have this.
name: Route
````

crdName: routes.route.openshift.io
featureGate: ExternalRouteCertificate
tests:
```

The integration tests use the crdName and featureGate to determine which tests apply to which manifests and automatically
react to changes when the FeatureGates are enabled/disabled on various FeatureSets and ClusterProfiles.

> [gen-minimal-test.sh](tests%2Fhack%2Fgen-minimal-test.sh) can still function to stub out files if you don't want to

Reviewer: ```suggestion
[`gen-minimal-test.sh`](tests/hack/gen-minimal-test.sh) can still function to stub out files if you don't want to
```
...although surprisingly GitHub seems equally happy to serve up both https://github.com/openshift/api/blob/ca1cfcd7c0e515141b98205f394b7c3f006ecb40/tests/hack/gen-minimal-test.sh and https://github.com/openshift/api/blob/ca1cfcd7c0e515141b98205f394b7c3f006ecb40/tests%2Fhack%2Fgen-minimal-test.sh, so the percent-encoding is unnecessary but doesn't actually break the link.

participants are aware the team is moving to that model.  If you'd like to sign your team up, please speak with Gina Hargan who will
be happy to help on-board your team.

## vendoring generated manifests into other repositories
If your repository relies on vendoring and copying CRD manifests (good job!), you'll need have an import line that
depends on the package that contains the CRD manifests.
For example, adding
```go
import (
	_ "github.com/openshift/api/operatoringress/v1/zz_generated.crd-manifests"
)
```

> Once added, a `go mod vendor` will pick up the package containing the manifests for you to copy.

Reviewer: Maybe worth mentioning that using `tools/tools.go` for these sort of build-time imports seems to be a common convention.
